### DISCUSSION TITLE
Fetch last git commit to the repo from Azure DevOps pipeline

### DISCUSSION THREAD

**Discussion by desaikeyurb** [community/community#140984] [Programming Help]
### Body

When my Azure DevOps yaml pipeline executes from master branch, I need to fetch last modified files/folders (basically last git commit to the repo). I found few commands like below, but none of them working in real execution of pipeline.

git diff $(Build.SourceVersion) --name-only --relative
git show --name-status
git diff --name-only HEAD~0 HEAD~1
git log -n --oneline
git log -p -1

Can anyone help here?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by VeerammalV** [comment]
trigger:
  branches:
    include:
      - master

pool:
  vmImage: 'ubuntu-latest'

steps:
  - checkout: self

  - script: |
      # Get the last commit ID
      LAST_COMMIT=$(git rev-parse HEAD)

      # Get the previous commit ID
      PREVIOUS_COMMIT=$(git rev-parse HEAD^)

      # Fetch last modified files
      echo "Last modified files:"
      git diff --name-only $PREVIOUS_COMMIT $LAST_COMMIT
    displayName: 'Get Last Modified Files'




This pipeline is triggered on commits to the master branch.
It checks out the code and retrieves the last and previous commit IDs.
It then lists the files modified between these two commits.

**Reply by desaikeyurb** [reply]
@VeerammalV - Thanks for the response. Tried creating test pipeline as you have suggested, however its giving me same error as below which I am getting for other Git commands:

**fatal: ambiguous argument 'HEAD^': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'**


"git rev-parse HEAD" command is working fine but not "git rev-parse HEAD^"

**Reply by desaikeyurb** [reply]
Can anyone help me with this? Any suggestions??

**Reply by flamableassassin** [reply]
Add [`fetchDepth: 0`](https://learn.microsoft.com/en-us/azure/devops/pipelines/yaml-schema/steps-checkout?view=azure-pipelines#shallow-fetch) to the checkout to include the history
